src/utils/components/SidebarEditor/utils.ts

```diff
@@ -8,3 +8,63 @@ export const safeLoad = <Resource>(value: string): Resource | undefined => {
     return;
   }
 };
+
+export type LineRange = { start: number; end: number };
+
+const getLineFromPath = (resourceYAML: string, path): LineRange => {
+  const yamlLines = resourceYAML.split('\n');
+  let startRange = 0;
+  let endRange = yamlLines.length - 1;
+
+  const properties = path.split('.');
+
+  for (const propertyDepth in properties) {
+    const property = properties[propertyDepth];
+
+    // at every iteration, go one level deeper, remove initial indentation for that range.
+    const replaceIndentationRegex = new RegExp(`^[ ]{${2 * parseInt(propertyDepth)}}`);
+
+    const rangeLines = yamlLines
+      .slice(startRange + 1, endRange)
+      .map((line) => line.replace(replaceIndentationRegex, ''));
+
+    // find the property
+    const startPropertyRange = rangeLines.findIndex((line) => line.startsWith(`${property}:`));
+
+    // find next property at same depth level
+    let rangeLength = rangeLines
+      .slice(startPropertyRange + 1)
+      .findIndex((line) => line.match(/^[A-z]+:/g));
+
+    if (rangeLength === -1) rangeLength = rangeLines.length - startPropertyRange;
+
+    // property not found
+    if (startPropertyRange === -1) return undefined;
+
+    startRange += startPropertyRange + 1;
+
+    endRange = startRange + rangeLength;
+  }
+
+  // editor lines starts from 1, array starts from 0
+  return { start: startRange + 1, end: endRange + 1 };
```

Comment:
can the + 1 be add in line 44, 46? here you can simply return ranges ? as suggested in line 17

Reply:
No, because all the previous calculations are done considering that the arrays start from 0, and only at the end, we need to add the +1.

If I do +1 on 44, and 46, I should also do some -1 in other places